src/ui/Search/frmsearchreplace.cpp

```diff
@@ -729,36 +730,49 @@ QStringList frmSearchReplace::fileFiltersFromUI()
     return filters;
 }
 
-void frmSearchReplace::addToHistory(QString string, QString type, QComboBox *comboBox)
-{
-    if (string != "") {
-        QSettings s;
-        QStringList history = s.value("Search/" + type + "History", QStringList()).toStringList();
-        history.prepend(string);
-        history.removeDuplicates();
-        history = history.mid(0, 10);
-        s.setValue("Search/" + type + "History", history);
-        comboBox->clear();
-        comboBox->addItems(history);
-    }
+void addToHistory(QStringList& history, QString string, QComboBox *comboBox) {
```

Comment:
Ok, now I see, you had to refactor the method because you couldn't pass a "setting name" as before. :+1: 

Can you add some docs over this method?